src/mlpack/methods/ann/layer/highway_impl.hpp

```diff
@@ -0,0 +1,249 @@
+
+  if (arma::size(output) != arma::size(input))
+  {
+    Log::Fatal << "The sizes of the output and input matrices of the Highway"
+        << " network should be equal. Please examine the network layers.";
+  }
+
+  transformGate = transformWeight * input;
+  transformGate.each_col() += transformBias;
+  transformGateActivation = 1.0 /(1 + arma::exp(-transformGate));
+  inputParameter = input;
+  networkOutput = output;
+  output = (output % transformGateActivation) +
+      (input % (1 - transformGateActivation));
+}
+
+template<typename InputDataType, typename OutputDataType,
+         typename... CustomLayers>
+template<typename eT>
+void Highway<InputDataType, OutputDataType, CustomLayers...>::Backward(
+    const arma::Mat<eT>&& /* input */,
+    arma::Mat<eT>&& gy,
+    arma::Mat<eT>&& g)
+{
+  boost::apply_visitor(BackwardVisitor(std::move(boost::apply_visitor(
+      outputParameterVisitor, network.back())),
+      std::move(gy % transformGateActivation),
+      std::move(boost::apply_visitor(deltaVisitor, network.back()))),
+      network.back());
+
+  for (size_t i = 2; i < network.size() + 1; ++i)
+  {
+    boost::apply_visitor(BackwardVisitor(std::move(boost::apply_visitor(
+        outputParameterVisitor, network[network.size() - i])), std::move(
+        boost::apply_visitor(deltaVisitor, network[network.size() - i + 1])),
+        std::move(boost::apply_visitor(deltaVisitor,
+        network[network.size() - i]))), network[network.size() - i]);
+  }
+
+  g = boost::apply_visitor(deltaVisitor, network.front());
+
+  transformGateError = gy % (networkOutput - inputParameter) %
+      transformGateActivation % (1.0 - transformGateActivation);
+  g += transformWeight.t() * transformGateError;
+  g += gy % (1 - transformGateActivation);
+}
+
+template<typename InputDataType, typename OutputDataType,
+         typename... CustomLayers>
+template<typename eT>
+void Highway<InputDataType, OutputDataType, CustomLayers...>::Gradient(
+    arma::Mat<eT>&& input,
+    arma::Mat<eT>&& error,
+    arma::Mat<eT>&& gradient)
+{
+  boost::apply_visitor(GradientVisitor(std::move(boost::apply_visitor(
+      outputParameterVisitor, network[network.size() - 2])),
+      std::move(error % transformGateActivation)), network.back());
+
+  for (size_t i = 2; i < network.size(); ++i)
+  {
+    boost::apply_visitor(GradientVisitor(std::move(boost::apply_visitor(
+        outputParameterVisitor, network[network.size() - i - 1])), std::move(
+        boost::apply_visitor(deltaVisitor, network[network.size() - i + 1]))),
+        network[network.size() - i]);
+  }
+
+  boost::apply_visitor(GradientVisitor(std::move(input), std::move(
+      boost::apply_visitor(deltaVisitor, network[1]))), network.front());
+
+  gradient.submat(0, 0, transformWeight.n_elem - 1, 0) = arma::vectorise(
+      transformGateError * input.t());
+  gradient.submat(transformWeight.n_elem, 0, gradient.n_elem - 1, 0) =
+      arma::sum(transformGateError, 1);
```

Comment:
Same here.

Reply:
No it is possible to fit them on the same line here.

Reply:
That comment was for @zoq :smile: